### PR TITLE
Authentication authorization adr

### PR DESCRIPTION
**Pull request by vahidbazzaz**
This PR contains authorization and authentication docs.

### CODE REVIEW

File: docs/adr/adr_authentication_auth0.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+## Decision Drivers
+
+1. Security / Risk for us
+2. Ease of Use
+3. Cost
+
+## Considered Options
+
+- Building our own authentication solution
```
Comment (aerinsol) [inline docs/adr/adr_authentication_auth0.md:21]: @vahidbazzaz didn't you have a bunch of other recommendations in your paper? I'm not sure I can find it in Slack now but I would either copy/paste or link it here.

File: docs/adr/adr_authorizations_auth0.md [hunk 2]
```diff
@@ -0,0 +1,223 @@
+
+# Authorization with Auth0
```
Comment (aerinsol) [inline docs/adr/adr_authorizations_auth0.md:2]: Do not overwrite the previous ADR with a completely new file, unless **we are talking specifically about the technology chosen for the authentication**. 

Architecture Decision Records are called **records** for a reason

File: docs/adr/adr_authorizations_auth0.md [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+
+## Considered Options
+
+- **Auth0 Core Authorization** (roles and permissions - rules)
+- **Auth0 Extended Authorization** (roles, permissions and groups - rules)
+- **Auth0 Organizations Authorization** (roles, permissions, organizations - rule)
+
+## Discussion
+
+The current system access control looks like this:
```
Comment (aerinsol) [inline docs/adr/adr_authorizations_auth0.md:32]: I think this is more clearly explained by a diagram

File: docs/adr/adr_authorizations_auth0.md (hunk 2 above)
Comment (vahidbazzaz) [reply docs/adr/adr_authorizations_auth0.md:2]: @aerinsol I actually renamed the file as we are mainly concerned with the Authentication part in this and the Authorization part in the other.

File: docs/adr/adr_authentication_auth0.md (hunk 1 above)
Comment (vahidbazzaz) [reply docs/adr/adr_authentication_auth0.md:21]: @aerinsol Sure, I'll update those links I missed

File: docs/adr/adr_authorizations_auth0.md (hunk 3 above)
Comment (vahidbazzaz) [reply docs/adr/adr_authorizations_auth0.md:32]: @aerinsol Yes, that makes perfect sense, I will try to create a diagram for that and link it to this, in fact, I have in mind to do so.

**Comment by vahidbazzaz** [comment]
@aerinsol I am closing this PR and creating two separate ones.

**Pull request closed without merging**